src/io/fabric8/Fabric8Commands.groovy

```diff
@@ -150,23 +150,50 @@ def getNewVersionFromTag(){
     return version
   }
 
-  tag = tag.trim()
-
-  // strip the v prefix from the tag so we can use in a maven version number
-  def previousReleaseVersion = tag.substring(tag.lastIndexOf('v')+1)
-  echo "Previous version found ${previousReleaseVersion}"
+  def semver = tag =~ /\bv?(?<major>0|[1-9]\d*)(?:\.(?<minor>0|[1-9]\d*)(?:\.(?<patch>0|[1-9]\d*))?)?(?:-(?<prerelease>[\da-z\-]+(?:\.[\da-z\-]+)*))?(?:\+(?<build>[\da-z\-]+(?:\.[\da-z\-]+)*))?\b/
+
+  if (semver.matches()) {
+    def majorVersion = semver.group('major') as int
+    def minorVersion = (semver.group('minor') ?: 0) as int
+    def patchVersion = ((semver.group('patch') ?: 0) as int) + 1
+
+    def pomSemver = pomVersion =~ /\bv?(?<major>0|[1-9]\d*)(?:\.(?<minor>0|[1-9]\d*)(?:\.(?<patch>0|[1-9]\d*))?)?(?:-(?<prerelease>[\da-z\-]+(?:\.[\da-z\-]+)*))?(?:\+(?<build>[\da-z\-]+(?:\.[\da-z\-]+)*))?\b/
+    if (pomSemver.matches) {
+      def pomMajorVersion = pomSemver.group('major') as int
+      def pomMinorVersion = (pomSemver.group('minor') ?: 0) as int
+      def pomPatchVersion = (pomSemver.group('patch') ?: 0) as int
+
+      if (pomMajorVersion > majorVersion) {
+        majorVersion = pomMajorVersion
+        minorVersion = pomMinorVersion
+        patchVersion = pomPatchVersion
+      } else if (pomMajorVersion == majorVersion && pomMinorVersion > minorVersion) {
+        minorVersion = pomMinorVersion
+        patchVersion = pomPatchVersion
+      } else if (pomMajorVersion == majorVersion && pomMinorVersion == minorVersion && pomPatchVersion > patchVersion) {
+        patchVersion = pomPatchVersion
+      }
```

Comment:
I see. So releases live in isolation, on no branch at all and master gets never updated (thats why you won't find release comments in the logs of 'master').